### PR TITLE
[Core][Geometries] Improve domain size computation in geometries

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

By suggestion of @pooyan-dadvand in some domain size (length, area, volume) it is possible to hardcode the calculus for the sake of efficiency. In here we allocate the vector and values when possible and compute directly if possible. I also refactor to reduce code duplication.

**🆕 Changelog**

- [Extend integration utilities](https://github.com/KratosMultiphysics/Kratos/commit/5a8c8cd9bb59f9285191ddfd6898c1c61a4ac731)
- [Typo in GeometryData](https://github.com/KratosMultiphysics/Kratos/commit/e1fb6c730b5d619a0d59f0d5397e172ba92ab022)
- [Minor improvement in base class](https://github.com/KratosMultiphysics/Kratos/commit/0cc0f44f146a6182546e4477ee6da86af41fa794)
- [Reduce code duplication with IntegrationUtilities](https://github.com/KratosMultiphysics/Kratos/commit/77dd3abd5dcd6ab442f4506ff0dd0825042bc963)
- [Reduce code duplication with IntegrationUtilities in quadratic lines](https://github.com/KratosMultiphysics/Kratos/commit/c4ced60b3c19ac3ef25c9c06b290d9ce1f6b6399)
- [Replace volume computation in prisms with IntegrationUtilities](https://github.com/KratosMultiphysics/Kratos/commit/03b823fa95c9ace85e16a0c1f94f306e1c169f8f)
- [Replace volume computation in pyramids with IntegrationUtilities](https://github.com/KratosMultiphysics/Kratos/commit/9068a0991a4906e7e9cd3b11b62950a247bf36d5)
- [Replace area computation in quadrilaterals with IntegrationUtilities](https://github.com/KratosMultiphysics/Kratos/commit/2dc0c7351b37a6e0281014fb5c3b7449070ed156)
- [Replace area computation in quadratic triangle with IntegrationUtilities](https://github.com/KratosMultiphysics/Kratos/commit/75c9f1386f17b0bde967e7c1120a83f1215fc473)

### CODE REVIEW

**Comment by loumalouomega** [comment]
This will be a draft until I fix some failing tests

**Comment by loumalouomega** [comment]
> This will be a draft until I fix some failing tests

~~~
Ran 1170 of 1173 test cases in 18.4243s (3 skipped). 9 failed:
    TestLengthLine2D3 Failed with message: 
        Error: Check failed because p_geometry->Length() = 1.02138 is not near to std::sqrt(2.0) = 1.41421 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_line_2d_3.cpp:103: virtual void Testing::TestLengthLine2D3::TestFunction()

    TestLengthLine3D3 Failed with message: 
        Error: Check failed because p_geometry->Length() = 1.02138 is not near to std::sqrt(2.0) = 1.41421 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_line_3d_3.cpp:104: virtual void Testing::TestLengthLine3D3::TestFunction()

    TestPrism3D15Length Failed with message: 
        Error: Check failed because geomRegular->Length() = 0.693361 is not near to 0.264567 = 0.264567 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_prism_3d_15.cpp:80: virtual void Testing::TestPrism3D15Length::TestFunction()

    TestPrism3D15Volume Failed with message: 
        Error: Check failed because geomRegular->Volume() = 9 is not near to 0.5 = 0.5 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_prism_3d_15.cpp:98: virtual void Testing::TestPrism3D15Volume::TestFunction()

    TestPrism3D6Length Failed with message: 
        Error: Check failed because geomRegular->Length() = 0.605707 is not near to 0.264567 = 0.264567 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_prism_3d_6.cpp:85: virtual void Testing::TestPrism3D6Length::TestFunction()

    TestPrism3D6Volume Failed with message: 
        Error: Check failed because geomRegular->Volume() = 6 is not near to 0.5 = 0.5 within the tolerance 1e-16
in kratos/tests/cpp_tests/geometries/test_prism_3d_6.cpp:104: virtual void Testing::TestPrism3D6Volume::TestFunction()

    TestPyramid3D13Volume Failed with message: 
        Error: Check failed because geomRegular->Volume() = 2.15083 is not near to 2.0 = 2 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_pyramid_3d_13.cpp:71: virtual void Testing::TestPyramid3D13Volume::TestFunction()

    TestPyramid3D5Volume Failed with message: 
        Error: Check failed because geomRegular->Volume() = 2.15083 is not near to 2.0 = 2 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_pyramid_3d_5.cpp:62: virtual void Testing::TestPyramid3D5Volume::TestFunction()

    TestTetrahedra3D10Volume Failed with message: 
        Error: Check failed because geom->Volume() = 4 is not near to 1.0 / 6.0 = 0.166667 within the tolerance 1e-06
in kratos/tests/cpp_tests/geometries/test_tetrahedra_3d_10.cpp:84: virtual void Testing::TestTetrahedra3D10Volume::TestFunction()
~~~

File: kratos/geometries/geometry.h [hunk 1]
```diff
@@ -3301,8 +3297,7 @@ class Geometry
 
         double detJ;
         Matrix Jinv(this->WorkingSpaceDimension(), this->WorkingSpaceDimension());
```
Comment (rubenzorrilla) [inline kratos/geometries/geometry.h]: Not related to the changes in here, but shouldn't this be
```suggestion
        Matrix Jinv(this->LocalSpaceDimension(), this->WorkingSpaceDimension());
```

Comment (loumalouomega) [reply kratos/geometries/geometry.h]: Nice catch!

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
Fine about optimizing. However, I think that in the particular case of geometries, it makes sense to keep the implementation without the refactoring. To be discussed in @KratosMultiphysics/technical-committee .

**Review by AlejandroCornejo — APPROVED** [review]
niceee

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee in agrees with the idea. However, there is doubt if this works correctly when the working space is larger than the local space (case of triangle 3d or quad 3d).

@loumalouomega  have you checked those cases?

**Comment by loumalouomega** [comment]
> @KratosMultiphysics/technical-committee in agrees with the idea. However, there is doubt if this works correctly when the working space is larger than the local space (case of triangle 3d or quad 3d).
> 
> @loumalouomega have you checked those cases?

Yes of course, in fact many of them are already implemented (with copy/paste) and already tested

**Review by rubenzorrilla — APPROVED** [review]
Then you can go ahead. Thanks!

**Pull request merged**